torch/optim/_muon.py

```diff
@@ -302,11 +302,8 @@ def _single_tensor_muon(
             raise ValueError("Param gradient must be a 2D matrix")
 
         buf = muon_momentum_bufs[i]
-        buf.mul_(momentum).add_(grad)
-        if nesterov:
-            grad = grad.add(buf, alpha=momentum)
-        else:
-            grad = buf
+        buf.lerp_(grad, 1 - momentum)
+        grad = grad.lerp_(buf, momentum) if nesterov else buf
```

Comment:
```suggestion
        grad = grad.lerp(buf, momentum) if nesterov else buf
```

don't inplace modify grads!

Reply:
right! updated